homeassistant/components/plaato/binary_sensor.py

```diff
@@ -46,7 +47,7 @@ def is_on(self):
             return self._coordinator.data.binary_sensors.get(self._sensor_type)
         return False
 
-    @property
+    @cached_property
     def device_class(self) -> BinarySensorDeviceClass | None:
         """Return the class of this device, from BinarySensorDeviceClass."""
         if self._coordinator is None:
```

Comment:
This can be set in the constructor in a separate PR as well

Reply:
Its a bit more complicated though since there are two methods here

Reply:
#100385